### ISSUE TITLE
Set initial learning rate

### ISSUE THREAD

**Issue by weka511** [weka511/learn#95]
I think my initial choice is not very good. There are some ideas in [How to Determine the Optimal Learning Rate of Your Neural Network](https://odsc.medium.com/how-to-determine-the-optimal-learning-rate-of-your-neural-network-42f5b59c5f01), including trying with a subset of data. The AI Overview has some interesting ideas:

- Train the model for a few epochs, starting with a very low learning rate (e.g., \(10^{-5}\)) and increasing it exponentially up to a high value (e.g., \(10\) or \(1\)). Plot the learning rate against training loss; the optimal rate is in the steepest downward slope.
- Use optimizers like [Adam](https://medium.com/data-science/how-to-choose-the-optimal-learning-rate-for-neural-networks-362111c5c783) or RMSprop that automatically adjust the learning rate during training, typically starting with a default of \(0.001\). _Maybe I don't need learning rate scheduler with Adam?_
-  Adam often works well with \(0.001\), while Stochastic Gradient Descent (SGD) may require higher initial values with decay.
- Larger batch sizes often allow for higher learning rates.
- Deeper networks might require lower, more cautious learning rates.
